packages/deployments/contracts/contracts/core/connext/facets/TokenFacet.sol

```diff
@@ -174,31 +179,47 @@ contract TokenFacet is BaseConnextFacet {
     address _stableSwapPool,
     uint256 _cap
   ) external onlyOwnerOrAdmin returns (address _local) {
+    // Calculate the canonical key.
+    bytes32 key = AssetLogic.calculateCanonicalHash(_canonical.id, _canonical.domain);
+
     bool onCanonical = _canonical.domain == s.domain;
-    if (!onCanonical) {
-      // On remote, deploy a local representation
-      _local = _deployRepresentation(
-        _canonical.id,
-        _canonical.domain,
-        _canonicalDecimals,
-        _representationName,
-        _representationSymbol
-      );
-    } else {
+    if (onCanonical) {
+      if (_cap > 0) {
+        // Set caps only on canonical domain
+        _setLiquidityCap(_canonical, _cap, key);
+      }
+
       // Get the local address
       _local = TypeCasts.bytes32ToAddress(_canonical.id);
 
       // You are on the canonical domain, ensure the adopted asset is empty
```

Comment:
document that `adopted` assets can only exist on non-canonical chains for a `canonical` token.